packages/x-tree-view-pro/src/internals/plugins/useTreeViewVirtualization/useTreeViewVirtualization.ts

```diff
@@ -0,0 +1,228 @@
+      virtualization: {
+        viewportHeight,
+        contentSize: itemCount * params.itemsHeight,
+      },
+    }));
+  });
+
+  const throttledUpdateDimensions = React.useMemo(
+    () => throttle(updateDimensions, params.resizeThrottleMs),
+    [params.resizeThrottleMs, updateDimensions],
+  );
+
+  const handleResizeRoot = () => {
+    const element = rootRef.current;
+    if (!element) {
+      return;
+    }
+
+    const computedStyle = ownerWindow(element).getComputedStyle(element);
+
+    const newSize = {
+      width: parseFloat(computedStyle.width) || 0,
+      height: parseFloat(computedStyle.height) || 0,
+    };
+
+    const prevDimensions = rootDimensionsRef.current;
+    rootDimensionsRef.current = newSize;
+
+    if (!prevDimensions || !areElementSizesEqual(prevDimensions, newSize)) {
+      // jsdom has no layout capabilities
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (newSize.height === 0 && !isJSDOM) {
+        emptyParentHeightWarning();
+      }
+
+      if (prevDimensions == null) {
+        // We want to initialize the Tree View dimensions as soon as possible to avoid flickering
+        updateDimensions();
+      } else {
+        throttledUpdateDimensions();
+      }
+    }
+  };
+
+  const isItemExpanded = instance.isItemExpanded;
+  const flatItemIds = React.useMemo(() => {
+    if (!experimentalFeatures.virtualization) {
+      return [];
+    }
+
+    const itemOrderedChildrenIds = state.items.itemOrderedChildrenIds;
+
+    const addChildrenToItem = (itemId: TreeViewItemId): TreeViewItemId[] => {
+      if (isItemExpanded(itemId)) {
+        return [itemId, ...(itemOrderedChildrenIds[itemId] ?? []).flatMap(addChildrenToItem)];
+      }
+
+      return [itemId];
+    };
+
+    return (itemOrderedChildrenIds[TREE_VIEW_ROOT_PARENT_ID] ?? []).flatMap(addChildrenToItem);
+  }, [state.items.itemOrderedChildrenIds, experimentalFeatures.virtualization, isItemExpanded]);
+
+  const computeRenderContext = React.useCallback(
+    (scrollPositionPx: number): UseTreeViewVirtualizationRenderContext => {
+      const clampItemIndex = (itemIndex: number) => clamp(itemIndex, 0, itemCount - 1);
+
+      const firstItemIndex = clampItemIndex(
+        Math.floor((scrollPositionPx - params.scrollBufferPx) / params.itemsHeight),
+      );
+
+      const lastItemIndex = clampItemIndex(
+        Math.ceil(
+          (scrollPositionPx + state.virtualization.viewportHeight + params.scrollBufferPx) /
+            params.itemsHeight,
+        ),
+      );
+
+      return {
+        firstItemIndex,
+        lastItemIndex,
+      };
+    },
+    [itemCount, params.itemsHeight, params.scrollBufferPx, state.virtualization.viewportHeight],
+  );
+
+  const getItemsToRenderWithVirtualization = (
+    renderContext: UseTreeViewVirtualizationRenderContext,
+  ) => {
+    const itemsToRender = flatItemIds.slice(
+      renderContext.firstItemIndex,
+      renderContext.lastItemIndex + 1,
+    );
```

Comment:
Can you add specific items?

Scroll makes you lose focus due to virtualization when the focused item is unmounted due to virtualization

Reply:
Indeed, I am not keeping the focused (and the edited) items rendered to avoid loosing some state information
It's one of the big improvements to add before opening for review

Reply:
I changed the behavior to keep the focused item but with its content hidden :+1: